timesketch/lib/analyzers/interface.py

```diff
@@ -136,14 +157,17 @@ def add_label(self, label, toggle=False):
         updated_event = self.datastore.set_label(
             self.index_name, self.event_id, self.event_type, self.sketch.id,
             user_id, label, toggle=toggle, single_update=False)
-        self._update(updated_event)
+        self.commit(updated_event)
```

Comment:
why have the commit here?

what if the analyzer does add_attributes more than once? wouldn't it be better to just update the `self.updated_event` and commit in the end?

Reply:
I can't update the event, because this is a special scripted update to set the label. It has to go in as it's own re-index event. I call self.commit(event) with the event and commit() will then just add it to it's queue and not touch self.updated_event.

Reply:
ack